hummingbot/connector/exchange/chainflip_lp/chainflip_lp_rpc_executor.py

```diff
@@ -0,0 +1,436 @@
+        if not response["status"]:
+            return []
+        return DataFormatter.format_balance_response(response["data"])
+    async def get_market_price(
+            self, base_asset:Dict[str, str],
+            quote_asset:Dict[str,str]):
+        params = {
+            "base_asset": base_asset,
+            "quote_asset": quote_asset
+        }
+        response = await self._execute_rpc_request(
+            CONSTANTS.MARKET_PRICE_V2_METHOD,
+            params
+        )
+        if not response["status"]:
+            return DataFormatter.format_error_response(response["data"])
+        return  DataFormatter.format_market_price(response["data"])
+    async def place_limit_order(
+            self, 
+            base_asset: Dict[str, str], 
+            quote_asset: Dict[str,str], 
+            order_id: str,
+            order_price: int,
+            side: Literal['buy'] | Literal['sell'], sell_amount: int):
+        tick = self._calculate_tick(
+            order_price, base_asset, quote_asset
+        )
+        if side == CONSTANTS.SIDE_BUY:
+            amount = DataFormatter.format_amount(sell_amount, quote_asset)
+        else:
+            amount = DataFormatter.format_amount(sell_amount, base_asset)
+        params = {
+            "base_asset": base_asset["asset"],
+            "quote_asset": quote_asset["asset"],
+            "id": order_id,
+            "side": side,
+            "tick":tick,
+            "sell_amount": amount
+        }
+        response = await self._execute_api_request(
+            CONSTANTS.PLACE_LIMIT_ORDER_METHOD,
+            params
+        )
+        if not response["status"]:
+            return DataFormatter.format_error_response(response["data"])
+        return DataFormatter.format_order_response(response["data"])
+    async def cancel_order(
+            self, base_asset: Dict[str,str], quote_asset: Dict[str,str], 
+            order_id: str, side: Literal['buy'] | Literal['sell']) -> bool:
+        params = {
+            "base_asset": base_asset["asset"],
+            "quote_asset": quote_asset["asset"],
+            "id": order_id,
+            "side": side,
+            "sell_amount": DataFormatter.format_amount(0, base_asset)
+        }
+        response = await self._execute_api_request(
+            CONSTANTS.CANCEL_LIMIT_ORDER,
+            params
+        )
+        return response["status"]
+
+    async def listen_to_market_price_updates(self, events_handler: Callable, market_symbol: str):
+        all_assets = await self.all_assets()
+        if not all_assets:
+            self.logger().error(
+                    f"Unexpected error getting assets from chainflip rpc. Error: {e}",
+                    exc_info=True
+                )
+            sys.exit()
+        while True:
+            try:
+                asset = DataFormatter.format_trading_pair(market_symbol, all_assets)
+                prices = await self.get_market_price(asset["base_asset"], asset["quote_asset"])
```

Comment:
there is the `subscribe_pool_price_v2` method that might be useful here

Reply:
The thing is that most of the RPC methods are not in the official documentation and anytime i see a method from the polkadot rpc explorer , i have to do a lot of trial and error to figure how to properly call the methods. i hope this issue is tended to soon. I will look into this method and see how it can be of use.

Reply:
i agree that the rpc documentation is not great yet unfortunately 😕 but this method is documented here: https://docs.chainflip.io/lp/integrations/lp-rpcs#cf_subscribe_pool_price_v2

Reply:
Please what do you use for your subscription because i tried using websocket connections and it is not going well

Reply:
websockets should be the correct approach. what error do you see?